### PR TITLE
[Doc Repo] Ensure file exists in DB before serving (Redmine #12220)

### PR DESCRIPTION
**Pull request by johnsaigle**
This pull request ensures that a requested file exists in the `document_repository` database before serving it.

- [x] Check that a file (in the format of `$username/$filename`) exists before serving

-  _This format is consistent w.r.t. the DB variable `Data_dir`. See [here](https://github.com/aces/Loris/blob/01f2467f9e150e617b1c6c9a0b7401edfdc59b1f/modules/document_repository/ajax/documentEditUpload.php#L63)._

- [x] Give a status code `400 Bad Request` otherwise
- [x] Fix a security issue regarding file access

See also: Redmine `#12220`

### CODE REVIEW

**Comment by jstirling91** [comment]
Whats still required for this pull request?

**Comment by johnsaigle** [comment]
## tl;dr: There might be a better fix, but we can merge this now if deadlines are tight!

@alisterdev and I found that the same issue exists in the media module. (see #2745)

However, his PR contains a different, shorter fix. I haven't had time to look over his solution to see whether it's better. 

I also think it would be wise to [force the file download](https://github.com/aces/Loris/pull/2745/files#diff-cfb2a308f601a2ab52b96eb34f8d322dR36) like Alex has. This would prevent issues that arise when we directly serve the file.

I'd like to have more time to look into the comparison of both approaches and verify whether his fix works. If so, I'd probably alter this PR to match and also force the download. That said, this PR works just fine as is, so if it's pressing we should just merge it and I'll compare the approaches another time.

**Comment by johnsaigle** [comment]
The code has been updated and is ready for merging.

At some point we should change the doc repo behaviour so access the user's name using the library classes instead of passing it through the path. However, it's good for now.

File: modules/document_repository/ajax/GetFile.php [hunk 1]
```diff
@@ -35,27 +34,33 @@
 // Checks that config settings are set
 $config =& NDB_Config::singleton();
 
-$File = $_GET['File'];
+$file = $_GET['File'];
```
Comment (alisterdev) [inline modules/document_repository/ajax/GetFile.php]: @johnsaigle Do you want to keep using the DB approach or switch to use `basename` for consistency?

Comment (johnsaigle) [inline modules/document_repository/ajax/GetFile.php]: `basename` won't work here because for whatever reason this module is coded to retrieve files by "`$user/$filename`" instead of `$filename`.

I do ultimately think that the `basename` approach is better but it would involve a more in-depth refactoring of this module.
